analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -176,6 +176,15 @@ class GoMod(
             graph = graph.subgraph(usedPackages)
         }
 
+        val referencedPackages = graph.getTransitiveDependencies(graph.projectId())
```

Comment:
To me it seems that this is a workaround for a bug in `subgraph()` and should have been fixed in the `subgraph()` function instead.

It's unfortunate that none of the GoMod tests reflect this fix, this would help to prevent regressions. If you know a case where this fix helps, would you mind adding it to the test project in a previous commit?

Reply:
Why do you think it's a bug in subgraph?
(I believe it isn't because it creates an overlay graph containing only the given set of nodes).

Reply:
How I understand the commit message is that if I have the graph A -> B -> C and call `subgraph(A, C)` it returns two loose graphs `A` and `C` even though it knows that `A` is the root node and `C` is not reachable from there anymore. That sounds very much like a bug to me. Btw, there is also a call to `subgraph` in line 110, doesn't it suffer from the same issue?

Reply:
> How I understand the commit message is that if I have the graph A -> B -> C and call `subgraph(A, C)` it returns two loose graphs `A` and `C` even though it knows that `A` is the root node and `C` is not reachable from there anymore. That sounds very much like a bug to me

To the best of my knowledge a graph doesn't have the concept of a root  node, so I cannot make sense of this argumentation.

> Btw, there is also a call to subgraph in line 110, doesn't it suffer from the same issue?

I don't see how this could suffer from the same problem. Maybe I'm missing something?

Reply:
> To the best of my knowledge a graph doesn't have the concept of a root node, so I cannot make sense of this argumentation.

I think you can because you mention the concept of a root in your own commit message.

But after all, wouldn't it have been a much simpler solution to the issue to just call `collectDependencies()´ on the "all" scope to get the package list?

Anyway, I will approve this to unblock the fix.